### PR TITLE
Simple refactor, typo fixes, added logger

### PR DESCRIPTION
**Pull request by shdowtail**

### CODE REVIEW

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java [hunk 1]
```diff
@@ -151,7 +164,7 @@ public void remove() {
   private static int defaultCapacity = 16;
 
   /**
-   * Holds the base hash entries. if the capacity is 2^N, than the base hash
+   * Holds the base hash entries. if the capacity is 2^N, thn the base hash
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:167]: sill minor typo: I guess it's "then" ;-)

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java [hunk 2]
```diff
@@ -619,7 +634,7 @@ public boolean equals(Object o) {
 
       T v1 = this.get(key);
       T v2 = that.get(key);
-      if ((v1 == null && v2 != null) || (v1 != null && v2 == null) || (!v1.equals(v2))) {
+      if ((v1 == null && v2 != null) || (v1 != null && v2 == null) || (v1!= null && !v1.equals(v2))) {
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:637]: we also may be able to use `if (!Object.equals(v1, v2)) { return false };`, but I'd have to check the implementation ;-)

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java (hunk 1 above)
Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:167]: Two times the charm :')

**Review by Aminmalek — COMMENTED** [review]
.

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java [hunk 3]
```diff
@@ -329,8 +329,7 @@ public boolean containsKey(int key) {
    *         false otherwise.
    */
   public boolean containsValue(Object o) {
-    for (Iterator<T> iterator = iterator(); iterator.hasNext(); ) {
-      T object = iterator.next();
+    for (T object : this) {
```
Comment (Aminmalek) [inline bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:345]: can explain why you used enhanced for-loop in this situation?

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:345]: No other reason than just making it more compact

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java (hunk 2 above)
Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:637]: I'll keep that in mind

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java (hunk 3 above)
Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:345]: Though I can see it being an issue for some people if I'm being honest

Comment (JohannesLichtenberger) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:345]: I think it's great :-)

Comment (Aminmalek) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:345]: thank you

File: bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java [hunk 4]
```diff
@@ -301,7 +314,7 @@ public void clear() {
     // And setting all the <code>next[i]</code> to point at
     // <code>i+1</code>.
     for (int i = 1; i < this.capacity; ) {
-      next[i] = ++i;
+      next[i] = i + 1;
```
Comment (JohannesLichtenberger) [inline bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: This is wrong, as it doesn't increment `i` anymore.

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: What am I not seeing?

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: Ah right it's not "I equals", my bad

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: image
image

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: oh the increment isn't even present in the method

Comment (shdowtail) [reply bundles/sirix-core/src/main/java/org/sirix/utils/IntToObjectMap.java:317]: Gonna keep an eye out for silly mistakes

**Comment by JohannesLichtenberger** [comment]
@shdowtail thanks for your contributions so far. Hope you'll be able to deep dive into the code base and keep on contributing :-)

The `sirix-*` images in https://github.com/sirixdb/sirixdb.github.io/tree/master/images describe various parts of the system. Maybe it's already of some use, but I'll write a new technical doc about the architecture, using these images :-)

**Comment by shdowtail** [comment]
> @shdowtail thanks for your contributions so far. Hope you'll be able to deep dive into the code base and keep on contributing :-)
> 
> The `sirix-*` images in https://github.com/sirixdb/sirixdb.github.io/tree/master/images describe various parts of the system. Maybe it's already of some use, but I'll write a new technical doc about the architecture, using these images :-)

I certainly will. Currently taking a course to refresh my mind of few things though

**Pull request merged**